internal/s3client/uplink.go

```diff
@@ -28,37 +27,40 @@ type Uplink struct {
 func NewUplink(conf Config) (Client, error) {
 	client := &Uplink{conf}
 
-	defaultConfDir := fpath.ApplicationDir("storj", "uplink")
-	setupDir, err := filepath.Abs(defaultConfDir)
+	if client.conf.ConfigDir != "" {
+		fmt.Printf("Using existing uplink config at %s\n", client.conf.ConfigDir)
+		return client, nil
+	}
+
+	client.conf.ConfigDir = fpath.ApplicationDir("storj", "s3-client", "uplink")
+
+	// remove existing s3client uplink config so that
+	// we can create a new one with up-to-date settings
+	err := os.RemoveAll(client.conf.ConfigDir)
 	if err != nil {
 		return nil, UplinkError.Wrap(fullExitError(err))
 	}
-	validForSetup, _ := fpath.IsValidSetupDir(setupDir)
 
-	// uplink configuration doesn't exists
-	if validForSetup {
-		fmt.Printf(`No existing uplink configuration located at (%v)...
-		Creating uplink configuration with the following settings:
+	fmt.Printf(`Creating uplink configuration with the following settings:
+	"--config-dir: %s,"
 	"--non-interactive: true",
 	"--api-key: %s",
-	"--enc.key: %s",
-	"--satellite-addr: %s
+	"--enc.encryption-key: %s",
```

Comment:
do we want to print out the keys being used ? Seems like a possible security issue.

Reply:
i was on the fence about it.  i can remove this